plugin-statusnotifier/statusnotifierwidget.cpp

```diff
@@ -88,15 +139,121 @@ void StatusNotifierWidget::itemAdded(QString serviceAndPath)
     mServices.insert(serviceAndPath, button);
     layout()->addWidget(button);
     button->show();
+
+    // show/hide the added item appropriately and show mShowBtn if needed
+    connect(button, &StatusNotifierButton::titleFound, [this, button] (const QString &title) {
```

Comment:
I wonder if there will be a race condition. The DBus call to get `Title` happens in the constructor of StatusNotifierButton. Maybe the `titleFound` signal can be fired before this connection is established?

Reply:
I remember that I thought about it and concluded that it's safe but, being extremely busy right now, I don't remember my reasoning ;) Will discuss it with you later.

In the 4 weeks I've used it, no problem has happened, although that's not a good reason.

Thanks again!